File: 04_quicksort/python/04_quicksort.py
def quicksort(arr):
  if len(arr) < 2:
    return arr
  pivot = arr[0]
  minors = [i for i in arr[1:] if i <= pivot]
  bigger = [i for i in arr[1:] if i > pivot]
  return quicksort(minors) + [pivot] + quicksort(bigger)

list = [2,5,9,8,10,42,9,100]

print(quicksort(list))

File: 04_quicksort/typescript/04_quicksort.ts
/**
 * @param arr number[]
 * @returns number[]
 * 
 * It'll returns a sorted array.
 */
function quicksort(arr: number[]): number[] {
  if(arr.length < 2) return arr;

  const pivot = arr.shift()!;
  const minors = arr.filter((v)=>v <= pivot);
  const bigger = arr.filter((v)=>v > pivot);
  return [...quicksort(minors),pivot, ...quicksort(bigger)];
}

const list = [2,5,9,8,10,42,9,100];


console.log(quicksort(list));
